File: bigtangle-typescript/src/net/bigtangle/params/MainNetParams.ts
import { NetworkParameters } from '../core/NetworkParameters.js';
import { BigInteger } from '../core/BigInteger';

// Add missing imports or declarations
// You may need to adjust the import paths as needed
// import { createGenesis } from '../core/Genesis'; // Uncomment and adjust if createGenesis is exported from another file

// Declare missing constants if not imported
const ID_MAINNET = "org.bigtangle.production";

// Add class-level properties for maxTarget and maxTargetReward
let maxTarget: BigInteger;
let maxTargetReward: BigInteger;

 
export class MainNetParams extends NetworkParameters {
 
    public id: string;
    public addressHeader: number;
    public p2shHeader: number;
    public dumpedPrivateKeyHeader: number;
    public acceptableAddressCodes: number[];
   

    constructor() {
        super();
        // !!!this is initial value and used in genesis block hash, it can be
        // changed only for height
        maxTarget = new BigInteger("578960377169117509212217050695880916496095398817113098493422368414323410");
        // !!!this is initial value and used in genesis block hash, it can be
        // changed only for height
        maxTargetReward = new BigInteger("5789603771691175092122170506958809164960953988171130984934223684143236");

        this.dumpedPrivateKeyHeader = 128;
        this.addressHeader = 0;
        this.p2shHeader = 5;
        this.acceptableAddressCodes = [this.addressHeader, this.p2shHeader];

        this.packetMagic = 0xf9beb4d9;
        this.bip32HeaderPub = 0x0488B21E; // The 4 byte header that serializes in
                                          // base58 to "xpub".
        this.bip32HeaderPriv = 0x0488ADE4; // The 4 byte header that serializes in
                                           // base58 to "xprv"

        this.genesisPub = "03d6053241c5abca6621c238922e7473977320ef310be0a8538cc2df7ee5a0187c";

        this.permissionDomainname = ["0222c35110844bf00afd9b7f08788d79ef6edc0dce19be6182b44e07501e637a58"];
     
       
        this.id = ID_MAINNET;
        this.subsidyDecreaseBlockCount = 210000;
        this.spendableCoinbaseDepth = 100;

        this.dnsSeeds = [];
    

        this.addrSeeds = [];
 
        // seeds for servers

    
    }

    public getAddressHeader(): number {
        return this.addressHeader;
    }

    public getP2SHHeader(): number {
        return this.p2shHeader;
    }

    public getDumpedPrivateKeyHeader(): number {
        return this.dumpedPrivateKeyHeader;
    }

    public getAcceptableAddressCodes(): number[] {
        return this.acceptableAddressCodes;
    }

    public getUriScheme(): string {
        return "bigtangle";
    }

   
    public serverSeeds(): string[] {
        // Return an array of server seeds; adjust as needed
        return [];
    }

    public getOrderPriceShift(orderBaseTokens: string): number {
        // Return a default value; adjust as needed
        return 0;
    }

    public static get(): MainNetParams {
        return new MainNetParams();
    }
}

File: bigtangle-core/src/main/java/net/bigtangle/params/MainNetParams.java
/*******************************************************************************
 *  Copyright   2018  Inasset GmbH. 
 *  
 *******************************************************************************/
/*
 * Copyright 2013 Google Inc.
 * Copyright 2015 Andreas Schildbach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.bigtangle.params;


import com.google.common.collect.ImmutableList;

import net.bigtangle.crypto.pq.PQConstants;

/**
 * Parameters for the main production network on which people trade goods and
 * services.
 */
public class MainNetParams extends NetworkParameters {
 

    public MainNetParams() {
        super();
        addressHeader = 0;
        p2shHeader = 5;
        acceptableAddressCodes = new int[] { addressHeader, p2shHeader };

        packetMagic = 0xf9beb4d9L;
        bip32HeaderPub = 0x0488B21E; // The 4 byte header that serializes in
                                     // base58 to "xpub".
        bip32HeaderPriv = 0x0488ADE4; // The 4 byte header that serializes in
                                      // base58 to "xprv"

        genesisPub = "03d6053241c5abca6621c238922e7473977320ef310be0a8538cc2df7ee5a0187c";

        permissionDomainname = ImmutableList.of("0222c35110844bf00afd9b7f08788d79ef6edc0dce19be6182b44e07501e637a58");

        // PQ governance: ML-DSA-87 from genesis; dual suite after the
        // configured chain length (default: never). Mainnet genesis is legacy
        // EC; the suite table only gates the post-quantum proposer path.
        setPqSuiteActivationHeight(PQConstants.SUITE_ML_DSA_ONLY, 0);
        long dualActivation = PQConstants.dualActivationHeightFromProperty();
        if (dualActivation >= 0) {
            setPqSuiteActivationHeight(PQConstants.SUITE_CAT5_DUAL_1, dualActivation);
        }
     
       
        id = ID_MAINNET;
      
        spendableCoinbaseDepth = 100;

        dnsSeeds = new String[] { "enrtree://0000000000000000000000000000000000000000000000000000000000000000@seeds.bigtangle.org" };
       

    }

    public String[] serverSeeds() {
        return new String[] { "92.5.34.128:80", "43.132.208.9:80", "43.162.118.46:80" };

    }

    private static MainNetParams instance;

    public static synchronized MainNetParams get() {
        if (instance == null) {
            instance = new MainNetParams();
        }
        return instance;
    }


}
